akka-stream/src/main/scala/DocumentStage.scala

```diff
@@ -86,42 +91,58 @@ private[akkastream] class DocumentStage[T](
       private def onFailure(reason: Throwable): Unit = {
         val previous = last.flatMap(_._3)
 
-        killLast()
-
         err(previous, reason) match {
-          case Cursor.Cont(_)     => ()
-          case Cursor.Fail(error) => fail(out, error)
-          case Cursor.Done(_)     => completeStage()
+          case Cursor.Cont(_) =>
```

Comment:
Would prefer to keep doing nothing at this pointin this `Cont` case, with `killLast` called anyway before case handling

Reply:
done!

Reply:
Should indicated that then there is no need to have `killLast` in `stop` and `fail` which are called just after this

Reply:
well `stop` and `fail` are called from other sites as well (which is why I orginaly moved `killLast` in the `Cont` case)

Reply:
Let revert for now

Reply:
done!